dual_net.py

```diff
@@ -134,6 +134,40 @@ def should_log(i):
                     logger.add_summary(weight_summaries, global_step)
             self.save_weights()
 
+    def validate(self, tf_records, batch_size=128, logdir=None, init_from=None, num_steps=1000):
+        if logdir is None:
+            print("Need to run eval with a logdir")
+            return
+        if not logdir.endswith('_eval'):
+            logdir += "_eval"
+
+        with self.sess.graph.as_default():
+            input_tensors = preprocessing.get_input_tensors(
+                TRAIN_BATCH_SIZE, tf_records)
+            output_tensors = dual_net(input_tensors, TRAIN_BATCH_SIZE,
+                                      train_mode=False, **self.hparams)
+            train_tensors = train_ops(
+                input_tensors, output_tensors, **self.hparams)
+
+            # just run our cost tensors
+            validate_tensors = {k: train_tensors[k] for k in (
+                'policy_cost', 'value_cost', 'l2_cost',
+                'combined_cost')}
+            self.initialize_weights(init_from)
```

Comment:
there might be some weirdness here; if you redefine the same network (albeit with different inputs), then because we didn't put explicit names on the graph nodes, it'll continue autoincrementing the weight names.

So if you call train() and then eval(), then initialize_weights will reinitialize the weights in the train()-defined network, as opposed to the eval()-defined network. It's some pretty unfortunate global state in TF that you have to solve with different sessions.

Reply:
yeah; honestly i thought about this and figured that punting by having it be a separate subcommand rather than try to bolt it onto `main.py train` was simplest.  But it'd be nice to have this in our end-to-end local loop test.

Reply:
Yeah, having it as a separate command should avoid the above issues.